apis/teamspeak_api.php

```diff
@@ -370,6 +370,78 @@ public function stopServer($sid)
         return (object) $result;
     }
 
+    /**
+     * Suspends an existing virtual server
+     *
+     * @param int $sid The virtual server ID
+     * @return stdClass An object containing the request response
+     */
+    public function suspendServer($sid)
+    {
+        try {
+            // Build the parameters array
+            $api_params = [
+                'virtualserver_autostart' => 0
+            ];
+
+            // Fetch the virtual server instance, and modify it
+            $server = $this->getServer($sid);
+            $result = $server->modify($api_params);
+
+            // Stop the server instance
+            $this->stopServer($sid);
+
+            // Add a control variable to know if the API request has been sent successfully
+            if (empty($result['error'])) {
+                $result['status'] = true;
+            }
+        } catch (Exception $e) {
+            $result = [
+                'error' => $e->getMessage(),
+                'status' => false,
+                'code' => $e->getCode()
+            ];
+        }
+
+        return (object) $result;
+    }
+
+    /**
+     * Unsuspends an existing virtual server
+     *
+     * @param int $sid The virtual server ID
+     * @return stdClass An object containing the request response
+     */
+    public function unsuspendServer($sid)
+    {
+        try {
+            // Build the parameters array
+            $api_params = [
+                'virtualserver_autostart' => 1
+            ];
+
+            // Fetch the virtual server instance, and modify it
+            $server = $this->getServer($sid);
+            $result = $server->modify($api_params);
```

Comment:
So a couple things here.  First, the result of getServer() is an stdClass object and has no modify() method.  You are looking for $server->instance->modify(), but also getServer() may return an object with errors and no ->instance property so you should check for that and probably return that object if errors exist.  These same things of course also apply to the suspendServer() method.

One other thing I encountered is that the API did not allow me to modify the server while it was stopped, but only if I ran startServer() before this modify() call.